### PR TITLE
Do not detect the DML issues on all DDL types from DDLProcessor in GetDMLIssues()

### PR DESCRIPTION
**Pull request by priyanshi-yb**
https://yugabyte.atlassian.net/browse/DB-14204

We should not detect issues on CREATE MVIEW/VIEW queries coming from pgss to `GetDMLIssues` as we are already detecting them in PL/pgSQL section.

Fix: I am skipping the DDLs in the `GetDMLIssues()` function for which we have the Processor in the DDLProcessor. For now this should be enough for our requirement.

### CODE REVIEW

File: yb-voyager/src/queryissue/unsupported_ddls.go [hunk 1]
```diff
@@ -579,6 +579,30 @@ func (tid *TriggerIssueDetector) DetectIssues(obj queryparser.DDLObject) ([]issu
 	return issues, nil
 }
 
+// ==============VIEW ISSUE DETECTOR ======================
+
+type ViewIssueDetector struct{}
+
+func (p *ParserIssueDetector) NewViewIssueDetector() *ViewIssueDetector {
+	return &ViewIssueDetector{}
+}
```
[the remark this answers is not in the document]
Comment (priyanshi-yb) [reply yb-voyager/src/queryissue/unsupported_ddls.go]: Yes already made that change after merging my PR to this

File: yb-voyager/src/queryparser/ddl_processor.go [hunk 2]
```diff
@@ -808,9 +808,69 @@ func (c *CreateType) GetObjectName() string {
...
+
+func (v *ViewProcessor) Process(parseTree *pg_query.ParseResult) (DDLObject, error) {
+	viewNode, ok := getCreateViewNode(parseTree)
+	if !ok {
+		return nil, fmt.Errorf("not a CREATE VIEW statement")
+	}
+	view := View{
+		SchemaName: viewNode.ViewStmt.View.Schemaname,
+		ViewName:   viewNode.ViewStmt.View.Relname,
+	}
```
Comment (sanyamsinghal) [inline yb-voyager/src/queryparser/ddl_processor.go:827]: A comment for later:
```
viewMsg := GetMessageField(parseTreeMsg, "create_view")
if viewMsg == nil {
}

view := View {
       SchemaName: GetStringField(viewMsg, "schemaname")
       ViewName: GetStringField(viewMsg, "relname")
}

```
This way no need to write a specific function just for views like `getCreateViewNode()`

File: yb-voyager/src/queryparser/ddl_processor.go [hunk 3]
```diff
@@ -808,9 +808,69 @@ func (c *CreateType) GetObjectName() string {
...
+func (mv *MViewProcessor) Process(parseTree *pg_query.ParseResult) (DDLObject, error) {
+	mviewNode, ok := getCreateTableAsStmtNode(parseTree)
+	if !ok {
+		return nil, fmt.Errorf("not a CREATE VIEW statement")
+	}
+	mview := MView{
+		SchemaName: mviewNode.CreateTableAsStmt.Into.Rel.Schemaname,
+		ViewName:   mviewNode.CreateTableAsStmt.Into.Rel.Relname,
+	}
+	return &mview, nil
```
Comment (sanyamsinghal) [inline yb-voyager/src/queryparser/ddl_processor.go:858]: similar comment to use protobuf msg

File: yb-voyager/src/queryparser/ddl_processor.go (hunk 2 above)
Comment (priyanshi-yb) [reply yb-voyager/src/queryparser/ddl_processor.go:827]: Yeah, will start utilising it as we use these processor.

Comment (priyanshi-yb) [reply yb-voyager/src/queryparser/ddl_processor.go:827]: Not doing these changes as of now.

File: yb-voyager/src/queryparser/helpers_struct.go [hunk 4]
```diff
@@ -288,3 +288,15 @@ func GetFuncParametersTypeNames(parseTree *pg_query.ParseResult) []string {
...
+func IsDDL(parseTree *pg_query.ParseResult) (bool, error) {
+	ddlParser, err := GetDDLProcessor(parseTree)
+	if err != nil {
+		return false, fmt.Errorf("error getting a ddl parser: %w", err)
+	}
+	_, ok := ddlParser.(*NoOpProcessor)
+	//Considering all the DDLs we have a Processor for as of now.
+	//Not Full-proof as we don't have all DDL types but atleast we will skip all the types we know currently
+	return !ok, nil
+}
```
Comment (sanyamsinghal) [inline yb-voyager/src/queryparser/helpers_struct.go:302]: Comment for later: I think if we can detect either - `isDDL` or `isDML` then we know...

IMO detecting DML is easier, only 4 types of statements/nodes - SELECT, INSERT, UPDATE, DELETE

Rest everything we can call as DDL
what do you think?

**Review by sanyamsinghal — APPROVED** [review]

**Pull request merged**